static/gsApp/views/spendCaps/spendCapSettings.tsx

```diff
@@ -0,0 +1,728 @@
+          onUpdate={handleUpdate}
+          reserved={null}
+        />
+        <SharedSpendCapPriceTable
+          activePlan={activePlan}
+          currentReserved={currentReserved}
+          additionalProducts={additionalProducts}
+        />
+      </Fragment>
+    );
+  }
+
+  return (
+    <InnerContainer>
+      <div>
+        <Subtitle>
+          {tct('Set [budgetMode] monthly spending caps', {
+            budgetMode: formattedBudgetMode,
+          })}
+        </Subtitle>
+        <Subtext>
+          {t(
+            'Set a maximum monthly spend to control costs. If you reach this limit, data collection pauses until the next billing cycle.'
+          )}
+        </Subtext>
+      </div>
+      <ExampleContainer>
+        <strong>{t('Example:')}</strong>
+        <p>
+          {tct(
+            "With a $300 cap, if you use $150 in [budgetTerm] beyond your included volumes, you'll be charged exactly $150. Your service continues normally until you hit the $300 limit.",
+            {budgetTerm}
+          )}
+        </p>
+      </ExampleContainer>
+      {inputs}
+      <StartingRate
+        alignSelf={
+          onDemandBudgets.budgetMode === OnDemandBudgetMode.PER_CATEGORY ? 'end' : 'start'
+        }
+      >
+        {t('* starting rate')}
+      </StartingRate>
+    </InnerContainer>
+  );
+}
+
+function BudgetModeSettings({
+  activePlan,
+  onDemandBudgets,
+  onUpdate,
+}: BudgetModeSettingsProps) {
+  const hasBudgetModes = activePlan.hasOnDemandModes;
+
+  if (!hasBudgetModes) {
+    return null;
+  }
+
+  return (
+    <StyledGrid columns="repeat(2, 1fr)" gap="xl">
+      {Object.values(OnDemandBudgetMode).map(budgetMode => {
+        const budgetModeName = capitalize(budgetMode.replace('_', '-'));
+        const isSelected = onDemandBudgets.budgetMode === budgetMode;
+        const nextOnDemandBudget = convertOnDemandBudget(onDemandBudgets, budgetMode);
+        return (
+          <BudgetMode
+            key={budgetMode}
+            isSelected={isSelected}
+            onClick={() => {
+              onUpdate({
+                onDemandBudgets: nextOnDemandBudget,
+              });
+            }}
+          >
+            <strong>
+              {tct('[budgetMode] spending caps', {
+                budgetMode: budgetModeName,
+              })}
+            </strong>
+            <StyledRadio
+              id={budgetMode}
+              name="budget-mode"
+              aria-label={`${budgetModeName} spending cap mode`}
+              value={budgetMode}
+              checked={isSelected}
+              readOnly
+            />
+          </BudgetMode>
+        );
+      })}
+    </StyledGrid>
+  );
+}
+
+function SpendCapSettings({
+  header,
+  activePlan,
+  onDemandBudgets,
+  onUpdate,
+  currentReserved,
+  isOpen,
+  additionalProducts,
+}: SpendCapSettingsProps) {
+  return (
+    <Flex direction="column">
+      {header}
+      {isOpen && (
+        <Fragment>
```

Comment:
lets use a grid with gap here to act as the layout. I noticed both `BudgetModeSettings` and `InnerSpendCapSettings` are using margin top to add spacing, we could have a grid here instead